curriculum/challenges/english/25-front-end-development/lab-bank-account-manager/6718d2d59337c822ecb697ff.md

```diff
@@ -0,0 +1,275 @@
+---
+id: 6718d2d59337c822ecb697ff
+title: Build a Bank Account Management Program
+challengeType: 14
+dashedName: build-a-bank-account-management-program
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should define a class named `BankAccount` with a constructor that takes `balance` as parameter. The constructor should set the default `balance` to `0`and initialize an empty array named `transactions` to store transaction records as objects. 
+
+2. Each transaction stored in the `transactions` array should be an object with two properties: `type` and `amount`. The `type` property should be either `deposit` or `withdraw`, and the `amount` property should be the amount deposited or withdrawn.
+
+3. Inside the `BankAccount` class, you should define a method named `updateBalance` that updates the balance according to the type of transaction. If the transaction type is `deposit`, it should add the amount to the balance. If the transaction type is `withdraw`, it should subtract the amount from the balance.
```

Comment:
```suggestion
3. Inside the `BankAccount` class, you should define a method named `updateBalance` that updates the balance using the `transactions` list, according to the type of transaction. If the transaction type is `deposit`, it should add the amount to the balance. If the transaction type is `withdraw`, it should subtract the amount from the balance.
```

I was really confused on what this method should do, there was no mention on using the list of all transactions to calculate the balance.

Reply:
The transactions is not a list, but there is a `transaction` array that has 2 types of transactions: `deposit` and `withdraw`.